datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -794,13 +794,20 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 op: Divide,
                 right,
             } if is_null(&right) => *right,
-            // A / A --> 1 (if a is not nullable)
+            // 0 / 0 -> null
```

Comment:
If we are returning an error for the A / 0 case, shouldn't we also return an error in this case? It seems like they are both divide-by-zero instances.

Reply:
I approached it a little bit complex than I should have. A / 0 is undefined however 0/0 is indeterminate so I represent undefined as error and indeterminate as null. Also, expected behavior in the #3615 pointed that `col / col` should return `null` when `col = 0`. I checked other languages and see that python implemented both (A/0 and 0/0) to return an error. Moreover, I like the offered solution. If it is OK for others, I am ready to change necessary parts. Thanks.

Reply:
> I think the expected behavior of a rewrite rule is the same behavior as the current expression evaluator.
> 
> One way to check the existing behavior is to use `datafusion-cli`. In this case the result of `0/0` is null:
> 
> ```shell
> $ datafusion-cli 
> DataFusion CLI v13.0.0
> ❯ select 0 / 0
> ;
> +---------------------+
> | Int64(0) / Int64(0) |
> +---------------------+
> |                     |
> +---------------------+
> ```

Thank you for your valuable feedback. If my understanding is wrong please correct me, we should return `null` for `0 / 0` to provide the same behavior with the current expression evaluator. So no changes are needed for this.

Reply:
> BTW if the behavior (error or NULL) is inconsistent in DataFusion I think we should file that as a separate ticket / PR rather than try to make it consistent in this one

I am a newbie rust developer and it is my first try to contribute to an open-source project. I am trying to understand the code base so I can't make appropriate comments on this one.

Reply:
It would be great if you can open an issue stating this discrepancy (A/0 -> Error vs. 0/0 -> NULL) so that the community can discuss/keep track of/fix it.

My understanding is that if you fix the lint issues and make the null-check (as in modulo), this could be good to go.

Reply:
We can follow-up it on #3849.

I did the necessary changes and control it with `cargo fmt`. Thanks for the help.